From 7d2af6d58d83115a9409d147893bff2a6ff7dac8 Mon Sep 17 00:00:00 2001
From: Ola Jeppsson <ola@adapteva.com>
Date: Thu, 27 Aug 2015 17:10:05 +0200
Subject: [PATCH 21/33] epiphany: Move frame pointer to register 15

2015-08-27  Ola Jeppsson  <ola@adapteva.com>

	* gcc/config/epiphany/epiphany.h: Move "fp" to register 15
	* gcc/config/epiphany/epiphany.md (GPR_FP): Change to 15

Signed-off-by: Ola Jeppsson <ola@adapteva.com>
---
 gcc/ChangeLog.epiphany          | 5 +++++
 gcc/config/epiphany/epiphany.h  | 8 ++++----
 gcc/config/epiphany/epiphany.md | 2 +-
 3 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/gcc/ChangeLog.epiphany b/gcc/ChangeLog.epiphany
index db91e0a3d2e..e75f10eb91d 100644
--- a/gcc/ChangeLog.epiphany
+++ b/gcc/ChangeLog.epiphany
@@ -1,3 +1,8 @@
+2015-08-27  Ola Jeppsson  <ola@adapteva.com>
+
+	* gcc/config/epiphany/epiphany.h: Move "fp" to register 15
+	* gcc/config/epiphany/epiphany.md (GPR_FP): Change to 15
+
 2014-12-12  Simon Cook  <simon.cook@embecosm.com>
 
 	* config/epiphany/epiphany.h (ENDFILE_SPEC): Add cachemanager.o
diff --git a/gcc/config/epiphany/epiphany.h b/gcc/config/epiphany/epiphany.h
index de877faa80b..483108cd5ec 100644
--- a/gcc/config/epiphany/epiphany.h
+++ b/gcc/config/epiphany/epiphany.h
@@ -715,11 +715,11 @@ typedef struct GTY (()) machine_function
 /* How to refer to registers in assembler output.
    This sequence is indexed by compiler's hard-register-number (see above).  */
 
-#define REGISTER_NAMES							\
-{									\
+#define REGISTER_NAMES						\
+{								\
   "r0",  "r1",  "r2",  "r3",  "r4",  "r5",  "r6",  "r7",	\
-  "r8",  "r9",  "r10", "fp",  "ip",  "sp",  "lr",  "r15",	\
-  "r16",  "r17","r18", "r19", "r20", "r21", "r22", "r23",	\
+  "r8",  "r9",  "r10", "r11", "ip",  "sp",  "lr",  "fp",	\
+  "r16", "r17", "r18", "r19", "r20", "r21", "r22", "r23",	\
   "r24", "r25", "r26", "r27", "r28", "r29", "r30", "r31",	\
   "r32", "r33", "r34", "r35", "r36", "r37", "r38", "r39",	\
   "r40", "r41", "r42", "r43", "r44", "r45", "r46", "r47",	\
diff --git a/gcc/config/epiphany/epiphany.md b/gcc/config/epiphany/epiphany.md
index a992d3c0795..aae463a18c2 100644
--- a/gcc/config/epiphany/epiphany.md
+++ b/gcc/config/epiphany/epiphany.md
@@ -23,10 +23,10 @@
 (define_constants
   [(GPR_0			 0)
    (GPR_1			 1)
-   (GPR_FP			11)
    (GPR_IP			12)
    (GPR_SP			13)
    (GPR_LR			14)
+   (GPR_FP			15)
    (GPR_16			16)
    (GPR_18			18)
    (GPR_20			20)
-- 
2.31.1

